src/extras/PMREMGenerator.js

```diff
@@ -104,6 +109,32 @@ PMREMGenerator.prototype = {
 
 	constructor: PMREMGenerator,
 
+	/**
+	 * Defines output and intermediary render target texture types.
+	 */
+	setDataType( type ) {
+
+		switch ( type ) {
+
+			case UnsignedByteType:
+				this._type = type;
+				this._format = RGBEFormat;
+				this._encoding = RGBEEncoding;
```

Comment:
I don't think this is enough granularity. Before it worked so that a JPG turns into `RGBFormat` with `sRGBEncoding`, while an HDR turns into `RGBEFormat` with `RGBEEncoding`. Not that JPG is the highest quality for PMREM, but it's fast, so long as you're not storing it as RGBE. Or is the idea that all JPGs should be decoded as Float? I suppose another option would be to have a data type = auto, which retains the old behavior.

Reply:
PMREM should output to linear color space. Right? Why sRGB?

Reply:
Well, it should be linear if it's float, but definitely non-linear if it's 8-bit. RGBE and sRGB are both nonlinear formats. If the input is sRGB, the output should match, unless it's upgraded to float.

Reply:
We need to separate encoding from color space.

>If the input is sRGB, the output should match

I see no reason for that.

I do not think PMREM should output to sRGB color space.

It is better to convert to linear once on the CPU rather than per pixel, per frame on the GPU.

Reply:
Yes, I agree, the problem is that floating point renderTargets aren't widely supported. In leu of that, it seems like converting sRGB is less onerous than converting RGBE, but maybe I'm wrong?

Reply:
That is exactly what I made initially.
If `setDataType( UnsignedByteType )` we modified to `RGBEEncoding`, otherwise we kept the input texture parameters.

Reply:
PMREM should always output to linear color space. It should also do its calculations in linear color space.

PMREM should output to RGBE if the output data type is `UnsignedByteType` and the input is .hdr or .exr. (maybe that is the point of confusion?).

RGBE is assumed to be in linear color space. We do not have an RGBE-sRGB encoding.